pkg/agent/utils.go

```diff
@@ -449,6 +488,7 @@ func getAKSKubeletConfiguration(kc map[string]string) *datamodel.AKSKubeletConfi
 		EnforceNodeAllocatable:         strings.Split(kc["--enforce-node-allocatable"], ","),
 		StreamingConnectionIdleTimeout: datamodel.Duration(kc["--streaming-connection-idle-timeout"]),
 		RotateCertificates:             strToBool(kc["--rotate-certificates"]),
+		ServerTLSBootstrap:             strToBool(kc["--rotate-server-certificates"]),
```

Comment:
will the flag always be set? iirc from the validator, this flag is set for 1.19 and above? what about provisioning for clusters < 1.19? i believe there are still some unsupported clusters out there, if they were to do SP reset, they'd get new logic and then this would fail?

IF they are using custom kubelet config. happy to leave this as is if you think we will always have the flag set, otherwise probably a good idea to first check if its present and then set it?

Reply:
I thought about this as well, though `strToBool` will return false if the flag isn't there (the implementation is quite... interesting)

and the RP validator now always explicitly sets the flag

Reply:
followed-up offline, will put in a fix to RP, ensuring we never set this flag on k8s < 1.8